simpleocr/files.py

```diff
@@ -10,80 +9,165 @@
-        else:
-            self.ground_path = basepath + GROUND_EXTENSIONS_DEFAULT
-            self.ground = None
+    def set_ground(self, segments, classes):
+        """ Creates the ground data in memory """
+        if self.is_grounded and self._debug:
+            print("Warning: grounding already grounded Image")
+        self._ground = Ground(segments=segments, classes=classes)
+
+    def remove_ground(self):
+        """ Removes the grounding data in memory for the Image """
+        if not self.is_grounded:
+            print("Warning: removing ground for Image without ground data")
+        self._ground = None
+
+    # These properties prevent the user from altering the attributes stored within
+    # the object and thus emphasize the immutability of the object
+    @property
+    def image(self):
+        return self._image
 
     @property
     def is_grounded(self):
-        """checks if this file is grounded"""
-        return not (self.ground is None)
+        return not (self._ground is None)
+
+    @property
+    def ground(self):
+        return self._ground
+
+
+class ImageFile(Image):
+    """
+    Complete class that contains functions for creation from file,
+    as well as from PIL Images. Also supports grounding in memory.
+    """
+    def __init__(self, path, debug=False):
+        """
+        :param path: path to the image to read, must be valid and absolute
+        :param debug: if True, debug prints are enabled
+        """
+        array = cv2.imread(path)
+        Image.__init__(self, array, debug=debug)
+        self._path = path
+        basepath = os.path.splitext(path)[0]
+        self._ground_path = try_extensions(GROUND_EXTENSIONS, basepath)
+        if self._ground_path:
+            self._ground = GroundFile(self._ground_path)
+            self._ground.read()
+        else:
+            self._ground_path = basepath + GROUND_EXTENSIONS_DEFAULT
+            self._ground = None
 
     def set_ground(self, segments, classes, write_file=False):
-        """creates the ground, saves it to a file"""
-        if self.is_grounded:
+        """ creates the ground, saves it to a file """
+        if self.is_grounded and self._debug:
             print("Warning: grounding already grounded file")
-        self.ground = GroundFile(self.ground_path)
+        self._ground = GroundFile(self._ground_path)
         self.ground.segments = segments
         self.ground.classes = classes
         if write_file:
             self.ground.write()
 
     def remove_ground(self, remove_file=False):
-        """removes ground, optionally deleting it's file"""
-        if not self.is_grounded:
+        """ Removes ground, optionally deleting it's file """
+        if not self.is_grounded and self._debug:
             print("Warning: ungrounding ungrounded file")
-        self.ground = None
+        self._ground = None
         if remove_file:
-            os.remove(self.ground_path)
+            os.remove(self._ground_path)
```

Comment:
It might make sense for ImageFile to expose the same internface as GroundFile (i.e.: implement `read` and `write`). A abc would be appropriate if you feel like doing it, but it's not too important